sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/data/OverviewViewportStateOptions.kt

```diff
@@ -0,0 +1,125 @@
+package com.mapbox.maps.plugin.viewport.experimental.data
+
+import com.mapbox.geojson.Geometry
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.plugin.viewport.DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS
+import java.lang.IllegalArgumentException
+import java.util.*
+
+/**
+ * Options that impact the [OverviewViewportState].
+ */
+class OverviewViewportStateOptions private constructor(
+  /**
+   * The geometry that the OverviewState should cover.
+   */
+  val geometry: Geometry,
+  /**
+   * The edge padding of the map.
+   */
+  val padding: EdgeInsets,
+  /**
+   * The bearing of the map.
+   */
+  val bearing: Double,
+  /**
+   * The bearing of the map.
+   */
+  val pitch: Double,
+  /**
+   * The maximum duration between frames in ms.
+   *
+   * Defaults to [DEFAULT_FRAME_TRANSITION_MAX_DURATION_MS] ms
+   */
+  val frameTransitionMaxDurationMs: Long
+) {
+  /**
+   * Returns a builder that created the [OverviewViewportStateOptions]
+   */
+  fun toBuilder() = Builder().geometry(geometry).padding(padding).bearing(bearing).pitch(pitch)
+    .frameTransitionMaxDurationMs(frameTransitionMaxDurationMs)
+
+  /**
```

Comment:
Are the comments to equals / hashCode / toString really needed? IMO just clutters the space

Reply:
These are public methods, our bot will complain if documentation for these methods are missing.

Reply:
Hmm, shouldn't we really just tweak the bot settings? Will save our time next times as well as code clarity and line numbers. Could you please hint what bot checks the comments?

Reply:
We use [this script](https://github.com/mapbox/mapbox-maps-android/blob/main/scripts/kdoc-validate.sh) to validate the missing API docs.